server/zio-http-server/src/main/scala/sttp/tapir/server/ziohttp/ZioHttpInterpreter.scala

```diff
@@ -31,62 +31,36 @@ trait ZioHttpInterpreter[R] {
     val zioHttpResponseBody = new ZioHttpToResponseBody
     val interceptors = RejectInterceptor.disableWhenSingleEndpoint(widenedServerOptions.interceptors, widenedSes)
 
-    def handleRequest(req: Request, filteredEndpoints: List[ZServerEndpoint[R & R2, ZioStreams with WebSockets]]) =
-      Handler.fromZIO {
-        val interpreter = new ServerInterpreter[ZioStreams with WebSockets, RIO[R & R2, *], ZioResponseBody, ZioStreams](
-          _ => filteredEndpoints,
-          zioHttpRequestBody,
-          zioHttpResponseBody,
-          interceptors,
-          zioHttpServerOptions.deleteFile
-        )
-
-        interpreter
-          .apply(ZioHttpServerRequest(req))
-          .foldZIO(
-            error => ZIO.fail(error),
-            {
-              case RequestResult.Response(resp) =>
-                resp.body match {
-                  case None              => handleHttpResponse(resp, None)
-                  case Some(Right(body)) => handleHttpResponse(resp, Some(body))
-                  case Some(Left(body))  => handleWebSocketResponse(body)
-                }
-
-              case RequestResult.Failure(_) =>
-                ZIO.fail(
-                  new RuntimeException(
-                    s"The path: ${req.path} matches the shape of some endpoint, but none of the " +
-                      s"endpoints decoded the request successfully, and the decode failure handler didn't provide a " +
-                      s"response. ZIO Http requires that if the path shape matches some endpoints, the request " +
-                      s"should be handled by tapir."
-                  )
-                )
-            }
+    HttpApp(
+      Routes.singleton {
```

Comment:
turns out with new zio-http you can simplify this whole thing and just remove the initial pre-filtering